pkg/ebpf/compiler/shim.h

```diff
@@ -0,0 +1,11 @@
+#ifndef __COMPILER_SHIM_H
+#define __COMPILER_SHIM_H
+
```

Comment:
Do you need 
```c++
#ifdef __cplusplus
extern "C" {
#endif
```

Reply:
This is included within the `extern "C"` block in the `.cpp` file, but this couldn't hurt.